### PR TITLE
feat: tickLabelFormat callback in axis component 

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**

### CODE REVIEW

File: src/plotContext.ts [hunk 1]
```diff
@@ -181,7 +184,7 @@ export function useAxisContext(
           axisContext[id] = {
             type: axis.scale,
             position: axis.position,
-            scientific: true,
+            tickLabelFormat: toExponential,
```
Comment (targos) [inline src/plotContext.ts:187]: We actually need to add the property to `PlotStateAxis` and use it like this:

```suggestion
            tickLabelFormat: axis.tickLabelFormat,
```

File: src/plotContext.ts [hunk 2]
```diff
@@ -141,6 +141,9 @@ interface SizeProps {
   plotHeight: number;
 }
 
+function toExponential(value: number) {
+  return value.toExponential(2);
+}
```
Comment (targos) [inline src/plotContext.ts:146]:
```suggestion
```

File: src/components/Axis/Axis.tsx [hunk 3]
```diff
@@ -44,7 +44,7 @@ export interface AxisProps {
   hiddenTicks?: boolean;
   tickPosition?: 'inner' | 'outer' | 'center';
   // TODO: Precise this.
-  tickLabelFormat?: () => string;
+  tickLabelFormat?: (any) => string;
```
Comment (targos) [inline src/components/Axis/Axis.tsx:47]:
```suggestion
  tickLabelFormat?: TickLabelFormat;
```

Same in other places, and add:

```ts
export type TickLabelFormat = (value: number) => string;
```

in `src/types.ts`

File: src/plotContext.ts [hunk 4]
```diff
@@ -193,7 +196,8 @@ export function useAxisContext(
           axisContext[id] = {
             type: 'linear' as const,
             position: axis.position,
-            scientific: diff <= 0.01 || diff >= 1000,
+            tickLabelFormat:
+              diff <= 0.01 || diff >= 1000 ? toExponential : undefined,
```
Comment (targos) [inline src/plotContext.ts:200]:
```suggestion
            tickLabelFormat: axis.tickLabelFormat,
```

**Comment by targos** [comment]
Please modify one of the stories to show that it works as expected.

For example in the Nasdaq story, we could have:
```js
const formatter = new Intl.DateTimeFormat('en-GB' , { timeStyle: 'short' });

<Axis
  label="Time (hh:mm)"
  tickLabelFormat={(time)=>timeFormat.format(time*1000)}

```

File: src/plotContext.ts (hunk 2 above)
Comment (wadjih-bencheikh18) [reply src/plotContext.ts]: fixed

File: src/components/Axis/types.ts [hunk 5]
```diff
@@ -20,5 +20,5 @@ export interface AxisRendererProps extends AxisCommonProps {
 }
 
 export interface AxisChildProps extends AxisCommonProps {
-  scientific: boolean;
+  tickLabelFormat?: (any) => string;
```
Comment (targos) [inline src/components/Axis/types.ts:23]:
```suggestion
  tickLabelFormat?: TickLabelFormat;
```

File: src/plotContext.ts [hunk 6]
```diff
@@ -48,7 +55,7 @@ interface PlotAxisContextGeneric<
   Scale extends ScaleContinuousNumeric<number, number>,
 > {
   scale: Scale;
-  scientific: boolean;
+  tickLabelFormat: (any) => string;
```
Comment (targos) [inline src/plotContext.ts:58]:
```suggestion
  tickLabelFormat: TickLabelFormat;
```

Comment (wadjih-bencheikh18) [reply src/plotContext.ts]: Fixed ,sorry forgot that

**Pull request merged**